docs/flow_filtering.md

```diff
@@ -0,0 +1,95 @@
+# eBPF Flow Rule Based Filtering
+
+## Introduction 
+
+Flow rule-base filtering is a method to control the flow of packets cached in eBPF's flows table based on certain configuration
+
+## Flow filter rule configuration
+
+ The Flow filter rule consists of two parts mandatory and optional parameters.
+ 
+### Mandatory parameters
+
+- `FLOW_FILTER_IP_CIDR` - IP address and CIDR mask for the flow filter rule, supports IPv4 and IPv6 address format.
+  If wanted to match against any IP, user can use `0.0.0.0/0` or `::/0` for IPv4 and IPv6 respectively.
+- `FLOW_FILTER_ACTION` - Action to be taken for the flow filter rule. Possible values are `Accept` and `Reject`.
+  - For the matching rule with `Accept` action this flow will be allowed to be cached in eBPF table, with updated global metric `FlowFilterAcceptCounter`.
+  - For the matching rule with `Reject` action this flow will be dropped and not cached in eBPF table, with updated global metric `FlowFilterRejectCounter`.
+  - If the rule is not matched, the flow will be allowed to be cached in eBPF table, with updated global metric `FlowFilterNoMatchCounter`.
+
+### Optional parameters
+
+- `FLOW_FILTER_DIRECTION` - Direction of the flow filter rule. Possible values are `Ingress` and `Egress`.
+- `FLOW_FILTER_PROTOCOL` - Protocol of the flow filter rule. Possible values are `TCP`, `UDP`, `SCTP`, `ICMP`, `ICMPv6`.
+- `FLOW_FILTER_SOURCE_PORT` - Single Source port of the flow filter rule.
+- `FLOW_FILTER_SOURCE_PORT_RANGE` - Source port range of the flow filter rule. using "80-100" format.
```

Comment:
assuming port and port-range are mutually exclusive?

Reply:
yes either one port or range but not both